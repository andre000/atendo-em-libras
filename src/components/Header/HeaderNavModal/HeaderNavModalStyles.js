import styled from 'styled-components'
import { PrimaryButton } from '../../Buttons'
import { Layer } from 'grommet'

const StyledButton = styled(PrimaryButton)`
  border-radius: 20px 20px 20px 0;
  background-color: white;
  color: #5996f7;
`

// height and position are handled manually to prevent
// fullscreen modal issue on mobile browsers. For more info read:
// https://css-tricks.com/the-trick-to-viewport-units-on-mobile/
const Modal = styled(Layer)`
  position: fixed;
  height: 100%;
  background-color: #5996f7;
`

const Logo = styled.img`
  height: auto;
  width: 100px;
`

const Wrapper = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: space-between;
  padding: 30px;
`

export { StyledButton, Modal, Logo, Wrapper }
