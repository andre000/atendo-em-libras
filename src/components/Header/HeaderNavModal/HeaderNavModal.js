import React, { PureComponent } from 'react'
import { Button } from '../../Buttons'
import { Box, Heading } from 'grommet'
import logo from '../../../assets/images/img_atendo_em_libras_logo_all_white.svg'
import { closeIcon } from '../../../assets/icons'
import { StyledButton, Modal, Logo } from './HeaderNavModalStyles'

class HeaderNavModal extends PureComponent {
  constructor(props) {
    super(props)
    this.myRef = React.createRef()
  }

  componentWillUnmount() {
    const node = this.myRef.current
    node.parentNode.removeChild(node)
  }

  render = () => {
    const { onClose } = this.props

    return (
      <Modal ref={this.myRef} modal plain full>
        <Box role="menu" flex="grow" justify="between" pad="large">
          <Box direction="row" justify="between">
            <Logo src={logo} alt={'Logo projeto Atendo em Libras'} />
            <Button
              aria-label="Close Menu Button"
              onClick={onClose}
              icon={<img src={closeIcon} alt="Close Icon" width="18px" height="18px" />}
              size="small"
            ></Button>
          </Box>
          <Box align="start">
            <Heading level="3" color="white" margin={{ bottom: 'medium' }}>
              É profissional de saúde ou conhece alguém da área?
            </Heading>

            <StyledButton
              href="https://forms.gle/h1sX9nD45PgUAzGB6"
              target="_blank"
              rel="noopener noreferrer"
              aria-label="register form"
              label="Cadastre-se aqui"
              size="medium"
            />
          </Box>
        </Box>
      </Modal>
    )
  }
}

export { HeaderNavModal }
