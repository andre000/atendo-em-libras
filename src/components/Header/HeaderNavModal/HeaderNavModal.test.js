import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { HeaderNavModal } from './HeaderNavModal'
import 'jest-styled-components'

describe('HeaderNavModal component', () => {
  let view
  let onClose

  beforeEach(() => {
    onClose = jest.fn()
    view = render(<HeaderNavModal onClose={onClose} />)
  })

  it('should execute callback function when close button is clicked', () => {
    const closeButton = view.getByLabelText('Close Menu Button')

    expect(closeButton).toBeInTheDocument()
    fireEvent.click(closeButton)
    expect(onClose).toHaveBeenCalled()
  })

  it('should redirect to form when register button is clicked', () => {
    const registerLink = view.getByLabelText('register form')

    expect(registerLink.href).toBe('https://forms.gle/h1sX9nD45PgUAzGB6')
  })

  it('Render correctly', () => {
    const { baseElement } = view
    expect(baseElement).toMatchSnapshot()
  })
})
