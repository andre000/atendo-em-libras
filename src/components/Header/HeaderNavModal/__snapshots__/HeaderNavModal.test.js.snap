// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavModal component Render correctly 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 48px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  padding-top: 96px;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c16 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c16:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c15 {
  margin-bottom: 24px;
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
  color: #FFFFFF;
}

.c13 {
  box-sizing: border-box;
  font-size: 18px;
  line-height: 24px;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
}

.c13:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: transparent;
  position: relative;
  z-index: 10;
  pointer-events: none;
  outline: none;
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
}

.c1 {
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  pointer-events: all;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 48px;
  outline: none;
  pointer-events: all;
  z-index: 15;
  position: absolute;
  max-height: calc(100% - 0px - 0px);
  max-width: calc(100% - 0px - 0px);
  border-radius: 0;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  -webkit-animation: gVgDGd 0.2s ease-in-out forwards;
  animation: gVgDGd 0.2s ease-in-out forwards;
}

.c4 {
  width: 0;
  height: 0;
  overflow: hidden;
  position: absolute;
}

.c9 {
  font-size: 1.4rem;
  font-weight: 600;
}

.c17 {
  background-color: #5996f7;
  border-radius: 17px;
  color: white;
  border-radius: 20px 20px 20px 0;
  background-color: white;
  color: #5996f7;
}

.c17:hover {
  background-color: #96bfff;
}

.c17:focus {
  outline: none;
  outline-offset: 0;
}

.c3 {
  position: fixed;
  height: 100%;
  background-color: #5996f7;
}

.c7 {
  height: auto;
  width: 100px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  margin-bottom: 40px;
}

.c12:last-child {
  margin-bottom: 0;
}

@media only screen and (max-width:768px) {
  .c5 {
    padding: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c10 {
    padding-top: 48px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    position: absolute;
    width: 100%;
    height: 100%;
    min-height: 100vh;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    position: relative;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    position: relative;
    max-height: none;
    max-width: none;
    border-radius: 0;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
    -webkit-animation: none;
    animation: none;
    height: 100vh;
    width: 100vw;
  }
}

<body>
  <div
    aria-hidden="false"
  />
  <div
    aria-hidden="false"
    id="layerClone"
  >
    <div
      aria-hidden="false"
    >
      <div
        class="c0"
        tabindex="-1"
      >
        <div
          class="c1"
        />
      </div>
    </div>
  </div>
  <div
    aria-hidden="false"
  />
  <div
    aria-hidden="false"
    id="layerClone"
  >
    <div
      aria-hidden="false"
    >
      <div
        class="c0"
        tabindex="-1"
      >
        <div
          class="c1"
        />
      </div>
    </div>
  </div>
  <div />
  <div
    aria-hidden="false"
    id="layerClone"
  >
    <div
      aria-hidden="false"
    >
      <div
        class="c0"
        tabindex="-1"
      >
        <div
          class="c1"
        />
      </div>
    </div>
  </div>
  <div />
  <div>
    <div
      aria-hidden="false"
    >
      <div
        class="c0"
        tabindex="-1"
      >
        <div
          class="c1"
        />
        <div
          class="c2 c3"
        >
          <a
            aria-hidden="true"
            class="c4"
            tabindex="-1"
          />
          <div
            class="c5"
            role="menu"
          >
            <div
              class="c6"
            >
              <img
                alt="Logo projeto Atendo em Libras"
                class="c7"
                src="img_atendo_em_libras_logo_all_white.svg"
              />
              <button
                aria-label="Close Menu Button"
                class="c8 c9"
                type="button"
              >
                <img
                  alt="Close Icon"
                  height="18px"
                  src="ic_close.svg"
                  width="18px"
                />
              </button>
            </div>
            <div
              class="c10"
            >
              <nav>
                <ol
                  class="c11"
                >
                  <li
                    class="c12"
                  >
                    <a
                      class="c13"
                      href="/"
                      theme="[object Object]"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    class="c12"
                  >
                    <a
                      aria-current="page"
                      class="c13 active"
                      href="/sobre"
                      style="color: rgb(89, 150, 247); border-bottom: 2px solid #5996f7;"
                      theme="[object Object]"
                    >
                      Sobre o Projeto
                    </a>
                  </li>
                </ol>
              </nav>
            </div>
            <div
              class="c14"
            >
              <h3
                class="c15"
              >
                É profissional de saúde ou conhece alguém da área?
              </h3>
              <a
                aria-label="register form"
                class="c16 c9 Buttons__PrimaryButton-nr8h4a-1 c17"
                href="https://forms.gle/h1sX9nD45PgUAzGB6"
                rel="noopener noreferrer"
                target="_blank"
              >
                Cadastre-se aqui
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
